### PR TITLE
[CN-248] Add PR-builder workflow

### PR DESCRIPTION
**Pull request by mtyazici**

### CODE REVIEW

File: .github/workflows/pr-builder.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            fi
+          done
+
+      - name: Lint Charts
+        run: |
+          ct lint --config ${GITHUB_WORKSPACE}/test/ct.yaml
+                    
+      - name: Create GKE cluster
+        run: |-
+          CLUSTER_NAME="charts-pr-test-${GITHUB_SHA::8}-${GITHUB_RUN_NUMBER}"
```
Comment (SeriyBg) [inline .github/workflows/pr-builder.yml:87]: Small neat: Maybe this one can be moved to the `env` section of the job?

**Review by SeriyBg — COMMENTED** [review]
Have just one small question about adding the git remote

File: .github/workflows/pr-builder.yml (hunk 1 above)
Comment (mtyazici) [reply .github/workflows/pr-builder.yml:90]: We cannot run `${GITHUB_SHA::8}` in the env section

File: .github/workflows/pr-builder.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      GCP_PROJECT_ID: hazelcast-33
+      GCP_NETWORK: operator-test-network
+      GKE_ZONE: europe-west1-b
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Add remote hazelcast
+        run: |
+          git remote add hazelcast https://github.com/hazelcast/charts.git
```
[the remark this answers is not in the document]
Comment (mtyazici) [reply .github/workflows/pr-builder.yml:24]: This is needed for PRs created from forks. Their origin is their fork and we need hazelcast repository to compare the charts.

File: .github/workflows/pr-builder.yml (hunk 1 above)
Comment (SeriyBg) [reply .github/workflows/pr-builder.yml:90]: Oh, I see. And the cluster name needs to be less than 40 chars

**Review by SeriyBg — APPROVED** [review]

File: .github/workflows/pr-builder.yml (hunk 1 above)
Comment (mtyazici) [reply .github/workflows/pr-builder.yml:90]: I was not aware of that one, I am not sure if it passes the limit. If not I can move it to env section by `           CLUSTER_NAME="charts-pr-test-${{ github.sha }}-${{ github.run_number}}"`

Comment (SeriyBg) [reply .github/workflows/pr-builder.yml:90]: I do have a limit of 40 chars (if I'm not mistaken it's 40). I've faced it few times 😃

**Comment by mtyazici** [comment]
Currently, `pull_request_target` workflow does not work because we don't have it in the main repository.

**Review by SeriyBg — APPROVED** [review]

**Review by hasancelik — APPROVED** [review]

**Pull request merged**